test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -176,59 +173,66 @@ jruby-config :- jruby-schemas/JRubyConfig
 (deftest test-ruby-http-client
   (jetty9/with-test-webserver ring-app port
```

Comment:
Where doest the value for the `port` come from?

Reply:
It gets filled and then yielded to the body by that [macro](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/blob/master/test/clj/puppetlabs/trapperkeeper/testutils/webserver.clj#L15-L16).

Reply:
Is [this](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/blob/master/test/clj/puppetlabs/trapperkeeper/testutils/webserver.clj#L34) what gets the actual value? Where is it getting the value from? I still find macros extremely confusing..

Reply:
Yes.

Reply:
Looks like we initialize the webserver right above that with a config that has `:port 0` in it. Then we ask the server for its post on that line.